src/dc_receive_imf.rs

```diff
@@ -4597,6 +4587,74 @@ Reply to all"#,
         }
     }
 
+    /// Tests that replies to similar ad hoc groups are correctly assigned to chats.
+    ///
+    /// The difficutly here is that ad hoc groups don't have unique group IDs, because both
+    /// messages have the same recipient lists and only differ in the subject and message contents.
+    /// The messages can be properly assigned to chats only using the In-Reply-To or References
+    /// headers.
+    #[async_std::test]
+    async fn test_chat_assignment_adhoc() -> Result<()> {
+        let alice = TestContext::new_alice().await;
+        let bob = TestContext::new_alice().await;
+        alice.set_config(Config::ShowEmails, Some("2")).await?;
+        bob.set_config(Config::ShowEmails, Some("2")).await?;
+
+        let first_thread_mime = r#"Subject: First thread
+Message-ID: first@example.org
+To: Alice <alice@example.com>, Bob <bob@example.net>
+From: Claire <claire@example.org>
+Content-Type: text/plain; charset=utf-8; format=flowed; delsp=no
+
+First thread."#
+            .as_bytes();
```

Comment:
You can put a `b` before the string (so you have `br#"Subject: ...`) and leave away the `as_bytes()` call, same with the second string below.